internal/server/instance/drivers/driver_lxc.go

```diff
@@ -2312,6 +2313,125 @@ func (d *lxc) startCommon() (string, []func() error, error) {
 		}
 	}
 
+	// Handle application containers.
+	if util.PathExists(filepath.Join(d.Path(), "config.json")) {
+		// Parse the OCI config.
+		data, err := os.ReadFile(filepath.Join(d.Path(), "config.json"))
+		if err != nil {
+			return "", nil, err
+		}
+
+		type ociConfig struct {
+			Process struct {
+				Terminal bool `json:"terminal"`
+				User     struct {
+					UID uint32 `json:"uid"`
+					GID uint32 `json:"gid"`
+				} `json:"user"`
+				Args []string `json:"args"`
+				Env  []string `json:"env"`
+				Cwd  string   `json:"cwd"`
+			} `json:"process"`
+		}
```

Comment:
Sooner than later you will want to deserialize into the upstream spec types ([runtime spec and/or image spec](https://github.com/opencontainers/image-spec/blob/v1.1.0/conversion.md#verbatim-fields)). They publish reference struct types in Go, and in implementations I've seen it's common to alias these

```go
import imagev1 "github.com/opencontainers/image-spec/specs-go"

type ImageManifest = imagev1.Manifest  // or maybe struct embedding
```

This will help with determining whether it's a bug in Incus or a malformed image.

Reply:
I'm not saying this struct in that repo. At least not when looking for the fields we're reading here.

Reply:
Forgive me, I mixed up image spec vs. runtime spec https://github.com/opencontainers/runtime-spec/blob/main/specs-go/config.go